packages/react-chat/src/views/ChatWindow/ChatWindowStandaloneView.tsx

```diff
@@ -0,0 +1,61 @@
+import { useContext, useEffect } from 'react';
+
+import { isObject } from '@/common/utils';
+import { RuntimeStateAPIContext, RuntimeStateContext } from '@/contexts';
+import { useTheme } from '@/hooks';
+import { noop } from '@/utils/functional';
+import { useResolveAssistantStyleSheet } from '@/utils/stylesheet';
+
+import { ChatAPI } from '../ChatWidget/types';
+import ChatWindow from '.';
+
+interface ChatWindowStandaloneViewProps extends React.PropsWithChildren {
+  chatAPI?: ChatAPI | undefined;
+  ready?: () => void;
+}
+
+const ChatWindowStandaloneView: React.FC<ChatWindowStandaloneViewProps> = ({ chatAPI, ready }) => {
+  const { shadowRoot } = useContext(RuntimeStateContext);
+  const { assistant, interact } = useContext(RuntimeStateAPIContext);
+  const theme = useTheme(assistant);
+
+  const isStyleSheetResolved = useResolveAssistantStyleSheet(assistant, shadowRoot);
+
+  useEffect(() => {
+    if (!isObject(chatAPI)) return undefined;
+    console.info('Methods open, close, hide, show, proactive.clear, proactive.push have no effect in this mode.');
+
+    Object.assign(chatAPI, {
+      open: noop,
+      close: noop,
+      hide: noop,
+      show: noop,
+      interact,
+      proactive: {
+        clear: noop, // () => setProactiveMessages([]),
+        push: noop, // (...messages: Trace.AnyTrace[]) => setProactiveMessages((prev) => [...prev, ...messages]),
+      },
+    });
+
+    ready?.();
+
+    return () => {
+      Object.assign(chatAPI, {
+        open: noop,
+        hide: noop,
+        show: noop,
+        close: noop,
+        interact: noop,
+        proactive: {
+          clear: noop,
+          push: noop,
+        },
+      });
+    };
+  }, []);
```

Comment:
could we use a common hook along with `ChatWidget` code [here](https://github.com/voiceflow/react-chat/blob/a3233a6e188504f10c5cf24cb325f8856e965968/packages/react-chat/src/views/ChatWidget/index.tsx#L33-L62)

And this component just overrides it? That way there isn't as much boilerplate, especially on the unmount